spyder/plugins/onlinehelp/widgets.py

```diff
@@ -78,123 +101,384 @@ def completer(self):
 
     def is_running(self):
         """Check if the server is running"""
-        if PY3:
-            if self.isRunning():
-                # Startup
-                return True
-            if self.server is None:
-                return False
-            return self.server.serving
-        else:
-            # Py 2
-            return self.isRunning()
+        if self.isRunning():
+            # Startup
+            return True
+
+        if self.server is None:
+            return False
+
+        return self.server.serving
 
     def quit_server(self):
         self.closed = True
         if self.server is None:
             return
-        if PY3:
-            # Python 3
-            if self.server.serving:
-                self.server.stop()
-        else:
-            # Python 2
-            if self.isRunning():
-                self.server.quit = 1
 
+        if self.server.serving:
+            self.server.stop()
 
-class PydocBrowser(WebBrowser):
-    """
-    pydoc widget
-    """
-    DEFAULT_PORT = 30128
 
-    def __init__(self, parent, options_button=None, handle_links=False):
-        WebBrowser.__init__(self, parent, options_button=options_button,
-                            handle_links=handle_links)
+class PydocBrowser(PluginMainWidget):
+    """PyDoc browser widget."""
+
+    DEFAULT_OPTIONS = {
+        'handle_links': False,
+        'max_history_entries': 10,
+        'port': 30128,

```

Comment:
Is there a need to add this to the widget options? We're not going to offer a UI to modify it, so don't we leave it as constant, like before?

Reply:
Because it is an option of the widget. Having it exposed as a configuration is not really relevant.

Reply:
To further elaborate, since this dict is used to store options/settings from the widgte, the option of leaving a constant is not appealing because:

`class PydocBrowser(PluginMainWidget): ``

1.) Having an attribute in the class means that now we have 2 different ways of handling options, some might be on the DEFAULTS_OPTIONS and some are class attributes. This is confusing and could enforce a bad practice.

2.) Having a constant outside the class might be better since now it is not a class attribute, so we are not incurring in 1, but now it lives outside the widget.

So I suggest we do not do that to enforce cleaner code, even if the option is not likely to change

Reply:
> some might be on the DEFAULTS_OPTIONS and some are class attributes

This is really not a class attribute. It's a constant, so why not simply declare it as such at the beginning of the file?